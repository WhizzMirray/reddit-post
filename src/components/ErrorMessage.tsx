import * as React from "react";

interface IErrorProps {
  message: string;
  clearError: () => void;
}

const ErrorMessage: React.SFC<IErrorProps> = props => (
  <div
    className="alert alert-danger"
    role="alert"
    style={{ marginBottom: "0px", padding: "8px 15px" }}
  >
    <span
      className="glyphicon glyphicon-exclamation-sign"
      aria-hidden="true"
      style={{ paddingRight: "5px" }}
    />
    <span className="sr-only">Error:</span>
    <span className="pull-right clickable">
      <button
        className="error-close btn btn-default"
        type="button"
        onClick={props.clearError}
      >
        <span className="glyphicon glyphicon-remove" />
      </button>
    </span>
    {props.message}
  </div>
);

export default ErrorMessage;