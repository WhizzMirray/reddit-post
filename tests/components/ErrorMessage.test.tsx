import { shallow } from "enzyme";
import * as React from "react";

import ErrorMessage from "../../src/components/ErrorMessage";

describe("<ErrorMessage/>", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<ErrorMessage message="hello" clearError={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it("should call mock function when the button is clicked", () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<ErrorMessage message="hello" clearError={mockFn} />);
    wrapper.find("button").simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
});
