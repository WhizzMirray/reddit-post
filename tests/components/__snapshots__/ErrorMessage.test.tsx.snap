// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage/> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    clearError={[MockFunction]}
    message="hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-exclamation-sign"
          style={
            Object {
              "paddingRight": "5px",
            }
          }
        />,
        <span
          className="sr-only"
        >
          Error:
        </span>,
        <span
          className="pull-right clickable"
        >
          <button
            className="error-close btn btn-default"
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="glyphicon glyphicon-remove"
            />
          </button>
        </span>,
        "hello",
      ],
      "className": "alert alert-danger",
      "role": "alert",
      "style": Object {
        "marginBottom": "0px",
        "padding": "8px 15px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "className": "glyphicon glyphicon-exclamation-sign",
          "style": Object {
            "paddingRight": "5px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error:",
          "className": "sr-only",
        },
        "ref": null,
        "rendered": "Error:",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="error-close btn btn-default"
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="glyphicon glyphicon-remove"
            />
          </button>,
          "className": "pull-right clickable",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="glyphicon glyphicon-remove"
            />,
            "className": "error-close btn btn-default",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "glyphicon glyphicon-remove",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        "type": "span",
      },
      "hello",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-exclamation-sign"
            style={
              Object {
                "paddingRight": "5px",
              }
            }
          />,
          <span
            className="sr-only"
          >
            Error:
          </span>,
          <span
            className="pull-right clickable"
          >
            <button
              className="error-close btn btn-default"
              onClick={[MockFunction]}
              type="button"
            >
              <span
                className="glyphicon glyphicon-remove"
              />
            </button>
          </span>,
          "hello",
        ],
        "className": "alert alert-danger",
        "role": "alert",
        "style": Object {
          "marginBottom": "0px",
          "padding": "8px 15px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "glyphicon glyphicon-exclamation-sign",
            "style": Object {
              "paddingRight": "5px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error:",
            "className": "sr-only",
          },
          "ref": null,
          "rendered": "Error:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="error-close btn btn-default"
              onClick={[MockFunction]}
              type="button"
            >
              <span
                className="glyphicon glyphicon-remove"
              />
            </button>,
            "className": "pull-right clickable",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="glyphicon glyphicon-remove"
              />,
              "className": "error-close btn btn-default",
              "onClick": [MockFunction],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "glyphicon glyphicon-remove",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "button",
          },
          "type": "span",
        },
        "hello",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
